src/widgets/UnwritableLocationErrorDialog.cpp

```diff
@@ -0,0 +1,102 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  UnwritableLocationError.h
+
+  Dmitry Vedenko
+
+**********************************************************************/
+
+#include "UnwritableLocationErrorDialog.h"
+
+#include "HelpSystem.h"
+#include "ShuttleGui.h"
+
+#include "prefs/PrefsDialog.h"
+#include "ui/AccessibleLinksFormatter.h"
+
+BEGIN_EVENT_TABLE(UnwritableLocationErrorDialog, wxDialogWrapper)
+    EVT_BUTTON(wxID_OK, UnwritableLocationErrorDialog::OnOk)
+    EVT_BUTTON(wxID_HELP, UnwritableLocationErrorDialog::OnError)
+END_EVENT_TABLE()
+
+UnwritableLocationErrorDialog::UnwritableLocationErrorDialog(wxWindow* parent, const wxString& path)
+    : wxDialogWrapper(
+         parent, -1, XO("Error"), wxDefaultPosition, wxDefaultSize, wxCAPTION | wxCLOSE_BOX)
+{
+   ShuttleGui S(this, eIsCreating);
+
+   S.SetBorder(8);
+
+   S.StartVerticalLay();
+   {
+      S.AddSpace(0, 12);
+
+      S.StartHorizontalLay();
+      {
+         S.AddSpace(12, 0);
+
+         S.StartVerticalLay();
+         {
+            S.AddFixedText(
+               /* i18n-hint: %s is replaced with a directory path. */
+               XO("Unable to write files to directory: %s.").Format(path),
+               false, 500);
+
+            S.AddFixedText(
+               /* i18n-hint: This message describes the error in the Error dialog. */
+               XO("Please check the directory exists, has the necessary permissions and the drive isn't full."),
```

Comment:
"Please check that the directory exists, has the necessary permissions, and the drive isn't full."

"that" and one "Oxford comma' improve the English a bit.